src/main/java/org/logstash/tcp/InputLoop.java

```diff
@@ -0,0 +1,125 @@
+package org.logstash.tcp;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import java.io.Closeable;
+
+/**
+ * Plain TCP Server Implementation.
+ */
+public final class InputLoop implements Runnable, Closeable {
+
+    /**
+     * Netty Boss Group.
+     */
+    private final EventLoopGroup boss;
+
+    /**
+     * Netty Worker Group.
+     */
+    private final EventLoopGroup worker;
+
+    /**
+     * The Server Socket's {@link ChannelFuture}.
+     */
+    private final ChannelFuture future;
+
+    /**
+     * Ctor.
+     * @param host Host to bind the listen to
+     * @param port Port to listen on
+     * @param decoder {@link Decoder} provided by Jruby
+     */
+    public InputLoop(final String host, final int port, final Decoder decoder) {
+        worker = new NioEventLoopGroup();
```

Comment:
This defaults to     
```
DEFAULT_EVENT_LOOP_THREADS = Math.max(1, SystemPropertyUtil.getInt(
                "io.netty.eventLoopThreads", NettyRuntime.availableProcessors() * 2));
```
_which delegates to  Runtime.getRuntime().availableProcessors()_

Is the *2 a concern here ?  I really have no idea what the proper number should be.